### DISCUSSION TITLE
How to set trigger to doing job repeatly with some interval after job is done

### DISCUSSION THREAD

**Discussion by hahyes** [quartznet/quartznet#1174] [Q&A]
Hi there,

another question from me. I'm trying to set trigger to repeat job forever with some interval (for example 5 seconds), but I want to start 5 seconds interval after job is done, not in meanwhile. Example I found in docs (this one in Quick Start), if job is taking more than 5 seconds, next start of job will be immediately. How to set trigger to make interval between jobs then? 

Cheers.

**Comment by lahma** [comment]
You should set [misfire policy,](https://www.quartz-scheduler.net/documentation/quartz-2.x/tutorial/simpletriggers.html#simpletrigger-misfire-instructions). You should probably decorate the job with `[DisallowConcurrentExecution]`. The docs should cover all this quite well.

**Reply by hahyes** [reply]
> You should set [misfire policy,](https://www.quartz-scheduler.net/documentation/quartz-2.x/tutorial/simpletriggers.html#simpletrigger-misfire-instructions).

Docs (at least from my understanding) states that misfire is not this kind of situation.

> A misfire occurs if a persistent trigger "misses" its firing time because of the scheduler being shutdown, or because there are no available threads in Quartz.NET's thread pool for executing the job.

For me it means that misfire is not related to situation, when trigger waits for finishing job with `[DisallowConcurrentExecution]`, that's why I didn't try to use it. Good to know that misfire policy is working in case like this, because docs is not that obvious about it.

Anyway misfire policy seems to not be a wanted solution for me sadly. I want to make breaks between runs of job, regular ones. Misfire instructions just reschedule job firing to next schedule time. But executing time of my jobs are not regular, so breaks between jobs will be irregular. 

However, I found solution in simple `await Task.Delay(TimeSpan.FromSeconds(x))` at the end of the job.

**Reply by lahma** [reply]
You also trigger the next run for trigger at the end of each run with desired delay and make initial configuration schedule it to run once.
